samples/Owin.Nowin.WebSockets/NowinServer.cs

```diff
@@ -27,9 +25,22 @@ public NowinServer(ServerBuilder serverBuilder)
             _serverBuilder = serverBuilder;
         }
 
-        public void Start(RequestDelegate requestDelegate)
+        public void Start<THttpContext>(IHttpApplication<THttpContext> application)
         {
-            _callback = requestDelegate;
+            _callback = async features =>
```

Comment:
Doesn't this have the same problem you had in WebListener? Does NOwin implement the OnStarting callback? (owin.OnSendingHeaders I think). That may not be called until after DisposeContext.